docs/upgrade.md

```diff
@@ -3,5 +3,263 @@ title: Upgrade guide
+
+=== "With AWS CDK"
+
+    ```typescript hl_lines="15 20-21"
+    import { Stack, type StackProps } from 'aws-cdk-lib';
+    import { Construct } from 'constructs';
+    import { NodejsFunction, OutputFormat } from 'aws-cdk-lib/aws-lambda-nodejs';
+    import { Runtime } from 'aws-cdk-lib/aws-lambda';
+
+    export class MyStack extends Stack {
+      public constructor(scope: Construct, id: string, props?: StackProps) {
+        super(scope, id, props);
+
+        const handler = new NodejsFunction(this, 'helloWorldFunction', {
+          runtime: Runtime.NODEJS_20_X,
+          handler: 'handler',
+          entry: 'src/index.ts',
+          bundling: {
+            format: OutputFormat.ESM,
+            minify: true,
+            esbuildArgs: {
+              "--tree-shaking": "true",
+            },
+            banner: 
+              "import { createRequire } from 'module';const require = createRequire(import.meta.url);",
+          },
+        });
+      }
+    }
+    ```
+
+=== "With AWS SAM"
+
+    ```yaml hl_lines="14 17-18"
+    Transform: AWS::Serverless-2016-10-31
+    Resources:
+      HelloWorldFunction:
+        Type: AWS::Serverless::Function
+        Properties:
+          Runtime: nodejs20.x
+          Handler: src/index.handler
+        Metadata:
+          BuildMethod: esbuild
+          BuildProperties:
+            Minify: true
+            Target: 'ES2020'
+            Sourcemap: true
+            Format: esm
+            EntryPoints:
+              - src/index.ts
+            Banner:
+              js: "import { createRequire } from 'module';const require = createRequire(import.meta.url);"
+ 
+    ```
+
+## Scoped imports
+
+### Middy.js middleware imports
+
+Code changes are required only if you're using Middy.js middlewares, however this update benefits especially those who are **not** using Middy.js middlewares due to less code being imported and bundled in your Lambda function.
+
+In v1, you could importing Middy.js middlewares in your codebase directly from the default export of a package. For example, to import the `injectLambdaContext` middleware for Logger, you would import it from `@aws-lambda-powertools/logger`. 
+
+With v2, we've added support for subpath exports. This means that you can now import Middy.js middlewares directly from a well-known path, i.e. `@aws-lambda-powertools/logger/middleware`. This allows you to import the middleware only when you need it, instead of requiring it and having it bundled in your Lambda function when you import the package.
```

Comment:
```suggestion
???+ note "Disregard if you are not using Middy.js middlewares."
    In v2, we've added support for subpath exports. This means if you don't import Middy.js middlewares, you will benefit from a smaller bundle size.

In v1, you could import Middy.js middlewares from the default export of a package _(e.g., `logger`)_. For example, you'd import `injectLambdaContext` Logger middleware from `@aws-lambda-powertools/logger`.

In v2, you can now import only the Middy.js middlewares you want to use from a subpath export, for example `@aws-lambda-powertools/logger/middleware`, leading to a smaller bundle size.
```

tip: bring the benefit/change upfront, then explain. you can shorten and even merge sentences by explaining certain topics earlier.